### PR TITLE
bug 1798741: escape path in source view

### PR DESCRIPTION
**Pull request by willkg**

### CODE REVIEW

File: webapp-django/crashstats/sources/tests/test_views.py [hunk 1]
```diff
@@ -109,3 +109,29 @@ def test_highlight_line(client, requests_mock):
...
+
+    url = reverse("sources:highlight_url")
+
+    # A missing line parameter means no lines are highlighted
+    response = client.get(url, {"url": f"https://{HOST}/foo<h1>html</h1>.c"})
+    assert response.status_code == 200
+    # Make sure the unescaped string doesn't show up at all
+    assert "<h1>html</h1>" not in smart_str(response.content)
+    # Make sure the escaped string does
+    assert "&lt;h1&gt;html&lt;/h1&gt;" in smart_str(response.content)
```
Comment (jwhitlock) [inline webapp-django/crashstats/sources/tests/test_views.py:137]: Nice confirmation of the escaping!

File: webapp-django/crashstats/sources/views.py [hunk 2]
```diff
@@ -72,8 +78,12 @@ def highlight_url(request):
             pass
 
     formatter = HtmlFormatter(
-        full=True, title=parsed.path, linenos="table", lineanchors="L", hl_lines=lines
+        full=True,
+        title=escape(parsed.path),
```
Comment (jwhitlock) [inline webapp-django/crashstats/sources/views.py:82]: 👍

**Review by jwhitlock — APPROVED** [review]
Thanks @willkg! This change looks good to me.

**Comment by willkg** [comment]
@jwhitlock Thank you for reviewing!

**Pull request merged**